src/Nethermind/Nethermind.Provide.Vault/VaultSealingHelper.cs

```diff
@@ -0,0 +1,112 @@
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Nethermind.Logging;
+using Nethermind.Vault.Config;
+using Nethermind.Vault.KeyStore;
+using Newtonsoft.Json;
+
+namespace Nethermind.Vault
+{
+    public class VaultSealingHelper : IVaultSealingHelper
+    {
+        private readonly IVaultKeyStoreFacade _vaultKeyStoreFacade;
+        private readonly IVaultConfig _config;
+        private readonly ILogger _logger;
+        public VaultSealingHelper(
+            IVaultKeyStoreFacade vaultKeyStoreFacade,
+            IVaultConfig config,
+            ILogger logger)
+        {
+            _vaultKeyStoreFacade = vaultKeyStoreFacade ?? throw new ArgumentNullException(nameof(vaultKeyStoreFacade));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+        public void Seal()
+        {
+            try
+            {
+                var sealResult = SealingUnsealingMethod("seal");
+                if (sealResult.Success)
+                {
+                    if (_logger.IsInfo) _logger.Info($"The vault sealing was successful.");
+                }
+                else
+                {
+                    if (_logger.IsError) _logger.Error($"The vault sealing was failed. {sealResult.Error}");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"Failed to seal vault", ex);
+            }
+        }
+
+        public void Unseal()
+        {
+            try
+            {
+                var unsealResult = SealingUnsealingMethod("unseal");
+                if (unsealResult.Success)
+                {
+                    if (_logger.IsInfo) _logger.Info($"The vault unsealing was successful.");
+                }
+                else
+                {
+                    if (_logger.IsError) _logger.Error($"The vault unsealing was failed. {unsealResult.Error}");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new ApplicationException($"Failed to unseal vault", ex);
+            }
+        }
+
+        class SealingUnsealingRequest
+        {
+            public string key { get; set; }
+        }
+        public (bool Success, string Error) SealingUnsealingMethod(string methodName)
+        {
+            var request = new SealingUnsealingRequest()
+            {
+                key = _vaultKeyStoreFacade.GetKey()
+            };
+
+            // with a new version of the Provide Nuget package we should remove httpClient call and use the provide.Unseal method
+            using (var httpClient = new HttpClient())
```

Comment:
This is invoked only once? If many times reuse of HttpClient is advised.

Reply:
Hmmm... two times, when the Vault plugin starts and when it is unloading. However, I skip that because I expect that the Provide add that method to Nuget soon, and we won't have to use httpClient anymore.